projects/plugins/social/src/js/components/panel/index.jsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Publicize sharing panel based on the
+ * Jetpack plugin implementation.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+	ConnectionVerify as PublicizeConnectionVerify,
+	Form as PublicizeForm,
+	useSocialMediaConnections as useSelectSocialMediaConnections,
+	usePostJustPublished,
+} from '@automattic/jetpack-publicize-components';
+import { __ } from '@wordpress/i18n';
+import { PanelBody, PanelRow, ToggleControl } from '@wordpress/components';
+import { store as editorStore } from '@wordpress/editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Description from './description';
+
+const PublicizePanel = ( { prePublish } ) => {
+	const { refresh, hasConnections, hasEnabledConnections } = useSelectSocialMediaConnections();
+	const isPostPublished = useSelect( select => select( editorStore ).isCurrentPostPublished(), [] );
+	const isPublicizeEnabled = useSelect(
+		select => select( 'jetpack/publicize' ).getFeatureEnableState(),
+		[]
+	);
+	const { togglePublicizeFeature } = useDispatch( 'jetpack/publicize' );
+
+	// Refresh connections when the post is just published.
+	usePostJustPublished(
+		function () {
+			if ( ! hasEnabledConnections ) {
+				return;
+			}
+
+			refresh();
+		},
+		[ hasEnabledConnections, refresh ]
+	);
+
+	// Panel wrapper.
+	const PanelWrapper = prePublish ? Fragment : PanelBody;
+	const wrapperProps = prePublish ? {} : { title: __( 'Share this post', 'jetpack-social' ) };
```

Comment:
Clever, I like it!

Reply:
I can't take the credit for that, it's in the Jetpack version too. I think that might have been @retrofox's handywork.